### PR TITLE
Docs: Update ecctl usage examples

### PR DESCRIPTION
**Pull request by kilfoyle**
I noticed that the [usage examples](https://www.elastic.co/guide/en/ecctl/current/ecctl-examples.html) section in the ecctl docs has only one example, and for me it isn't working.  So, I thought I'd take the examples from @Kushmaro's excellent [blog post](https://docs.google.com/document/d/1zrDBEWWM8HGmZ1vgSiG_Q8VOqxCpKgBjtrAA5k8qYGA/edit) and port them into the ecctl docs.  This way the ecctl examples can be very similar to the ones we'll have for the ESS Public API.

Unfortunately, I'm hitting an error with the `shutdown deployment` command.  

```
Davids-MacBook-Pro:testing.ecctl$ ecctl deployment shutdown $DEPLOYMENT_ID
This action will delete the specified deployment ID and its associated sub-resources, do you want to continue? [y/n]: y
{
  "errors": [
    {
      "code": "deployments.restricted_parameter",
      "fields": [
        "hide"
      ],
      "message": "Parameter [hide] is restricted and can only be set by a Platform administrator"
    }
  ]
}
```

@marclop and @karencfv , if you're the right people to ask, can you advise on the error above?  If `shutdown deployment` is no longer available as part of ecctl, perhaps in the docs I can just tell folks to use the Cloud UI to delete their their deployment.

### CODE REVIEW

**Comment by karencfv** [comment]
@kilfoyle, thank you so much for finding/reporting this error 🙏 .  Looks like the permissions for the `hide` parameter were recently changed :( I'll make a PR to address that issue today and report back.

**Comment by karencfv** [comment]
I've just tried the command myself and checked the code. The `hide` field is set to false and always has been, so it works for me and should work fine for you as well. Just to confirm, are you running the command against the public API? Are you building the command from source, or are you using a previous version of ecctl?

File: docs/ecctl-getting-started.asciidoc [hunk 1]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
 [id="{p}-examples"]
 == Usage examples
 
-To create your first deployment, use the `ecctl deployment create` command, which accepts https://github.com/elastic/ecctl/blob/master/docs/ecctl_deployment_create.md#examples[a deployment description in the form of a JSON file] and a stack version and name as inputs.
+Once you've <<{p}-installing,installed>> and <<{p}-configuring,configured>> {p} you're ready to take it for a test drive. Here are some examples for you to try, demonstrating basic Create, Read, Update and Delete operations using {p}:
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:255]:
```suggestion
Once you've <<{p}-installing,installed>> and <<{p}-configuring,configured>> {p} you're ready to take it for a test drive. Try these examples with basic Create, Read, Update and Delete operations using {p}:
```

File: docs/ecctl-getting-started.asciidoc [hunk 2]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
...
 
-[source,sh]
+* <<{p}-example-list-deployments,List deploymments>>
+* <<{p}-example-create-deployment,Create a deployment>>
+* <<{p}-example-update-deployment,Update a deployment>>
+* <<{p}-example-delete-deployment,Delete a deployment>>
+
+
+If you would like to compare the {p} commands against their API equivalents, see the {cloud}/ec-api-examples.html[API examples].
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:263]:
```suggestion
To compare the {p} commands against their API equivalents, see the {cloud}/ec-api-examples.html[API examples].
```

File: docs/ecctl-getting-started.asciidoc [hunk 3]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
...
+
+In this last example, you can use the <<ecctl_deployment_shutdown,{p} deployment shutdown>> command to delete the deployment that you created.
+
+[source, json]
+--
+ecctl deployment shutdown $DEPLOYMENT_ID
+--
+* `$DEPLOYMENT_ID` is the ID for the deployment that was created in the previous <<{p}-example-create-deployment,create a deployment>> example.
+
+On running this and other destructive commands, {p} prompts you with a confirmation message. You can use the `--force` option to skip the confirmation step, which can be useful if you are using {p} for automation.
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:537]:
```suggestion
On running this and other destructive commands, {p} prompts you with a confirmation message. Use the `--force` option to skip the confirmation step, if you are using {p} for automation.
```

File: docs/ecctl-getting-started.asciidoc [hunk 4]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
...
+
+There’s much more you can use {p} for. You can explore its various commands by issuing `ecctl <command> <help>` to see the different options that we support.
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:539]:
```suggestion
To see the different options that {p} supports, run `ecctl <command> <help>`.
```

File: docs/ecctl-getting-started.asciidoc [hunk 5]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
...
+      "kind": "apm",
+      "ref_id": "main-apm",
+      "region": "gcp-us-central1",
+      "secret_token": "zfufcfe15eCVJk78b5"
+    }
+  ]
+}
+--
+
+The response indicates that the request was submitted successfully. It includes the `elastic` user password, which you can use to log in to Kibana or to access the Elasticsearch REST API. Make a note of the deployment ID, which we will use in the next example.
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:432]:
```suggestion
The response indicates that the request was submitted successfully. It includes the `elastic` user password, which you can use to log in to Kibana or to access the Elasticsearch REST API. Make a note of the deployment ID, which you will use in the next example.
```

File: docs/ecctl-getting-started.asciidoc [hunk 6]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
...
+
+[id="{p}-example-update-deployment"]
+=== Update a deployment
+
+Now that you have used {p} to create a deployment, let's update it. We will scale it up, increasing the size of the Elasticsearch data nodes from 4096 to 8192 MB.
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:437]:
```suggestion
Now that you have used {p} to create a deployment, you can scale it up, by increasing the size of the Elasticsearch data nodes from 4096 to 8192 MB.
```

File: docs/ecctl-getting-started.asciidoc [hunk 7]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
...
+
+
+If you would like to compare the {p} commands against their API equivalents, see the {cloud}/ec-api-examples.html[API examples].
+
+
+
+[id="{p}-example-list-deployments"]
+=== List deployments
+
+As a first example of using {p}, let's run the <<ecctl_deployment_list,{p} deployment list>> command to retrieve information about existing deployments. This is a good way to confirm that {p} is configured correctly. If you have any deployments already created, they'll be listed in the results.
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:270]:
```suggestion
As a first example of using {p}, run the <<ecctl_deployment_list,{p} deployment list>> command to retrieve information about existing deployments. This is a good way to check if {p} is configured correctly and if you have any deployments already created.
```

File: docs/ecctl-getting-started.asciidoc [hunk 8]
```diff
@@ -252,47 +252,292 @@ https://golang.org/pkg/text/template/[`text/templates`] on demand.
...
+--
+ID                                 NAME            ELASTICSEARCH                      KIBANA                             APM                                APPSEARCH
+00be03849b6a49c1a6541e3ccb5958d2   marvin          00be03849b6a49c1a6541e3ccb5958d2   266e456acf257588a9cde6fb4569d4a0   78c096c22e12408b878083b2d5ff6bcf   -
+147cdeace6404c3e4b5018e1401647e4   biggerdata      147cdeace6404c3e4b5018e1401647e4   443a9df7b33952f45921c5823cbad4bc   4678ce52d45547e463455ede663cb4a4   -
+--
+
+[id="{p}-example-create-deployment"]
+=== Create a deployment
+
+Let's create a basic deployment. Elasticsearch Service supports {cloud}/ec-getting-started-templates.html[deployment templates], which pre-configure the Elastic Stack components in your deployment to best suit your particular use case. For this example, we'll use Google Cloud Platform (GCP) to host the deployment in region `US Central 1 (Iowa)`. For details about what deployment options are available by platform, see our list of {cloud}/ec-regions-templates-instances.html[available regions, deployment templates and instance configurations].
```
Comment (alaudazzi) [inline docs/ecctl-getting-started.asciidoc:287]:
```suggestion
Let's create a basic deployment. Elasticsearch Service supports {cloud}/ec-getting-started-templates.html[deployment templates], which pre-configure the Elastic Stack components in your deployment to best suit your particular use case. For this example, use Google Cloud Platform (GCP) to host the deployment in region `US Central 1 (Iowa)`. To know which deployment options are available by platform, see {cloud}/ec-regions-templates-instances.html[available regions, deployment templates and instance configurations].
```

**Comment by kilfoyle** [comment]
> I've just tried the command myself and checked the code. The `hide` field is set to false and always has been, so it works for me and should work fine for you as well. Just to confirm, are you running the command against the public API? Are you building the command from source, or are you using a previous version of ecctl?

Thanks for the suggestions @karencfv!  I am running the command against the public API on Staging.  I was using a previous version of ecctl but even after upgrading I hit the same problem.

Here's my version info:

```
Davids-MacBook-Pro:.ecctl$ brew list --versions elastic/tap/ecctl
ecctl 1.0.0-beta2
```

And the contents of ~/.ecctl/config.json:
```
{
  "host": "https://staging.found.no",
  "apikey": "<key deleted for github>",
  "output": "text",
  "timeout": 60000000000,
  "insecure": true
}
```

I retried listing and updating deployments, so that part still works fine, but  `ecctl deployment shutdown <deployment ID>` still fails. :-(

**Comment by kilfoyle** [comment]
Thanks for the text suggestions @alaudazzi !

**Comment by karencfv** [comment]
Ah! I see. The host for the public api is https://api.elastic-cloud.com, so you might want to have your ecl config like this:
```
host: https://api.elastic-cloud.com
region: us-west-2
apikey: <REDACTED>
```

**Comment by karencfv** [comment]
I'm guessing you used the `ecctl init` command to create your config file? The new version has a slightly different init flow in which ESS is supported :)

**Comment by kilfoyle** [comment]
Hi @karencfv ,

I'm sorry to report bad news, but I changed to using a config.yml file with the settings you have above, except of course with my own API key, and with the server set to ESS rather than Staging. I still have the same issue though with the `ecctl deployment shutdown $deploymentID` command.

./.ecctl/config.yml:
```
host: https://api.elastic-cloud.com
region: us-west-2
apikey: <redacted>
```

Would you perhaps have some time in the next week or so when we could have a look together to figure out what the issue is?

**Comment by karencfv** [comment]
I've just tried the command again and now it did not work for me 😄. Looks like this is a new change with the API that is being rolled out with ms36 (that's probably why the command didn't work in staging but it worked on prod a few days ago). 

I'm really sorry you had to go through all this @kilfoyle , but thank you for finding this bug for us before the v1.0.0-beta3 release! 🙇‍♀️ 

I'll make a PR with a fix later today or tomorrow.

**Comment by kilfoyle** [comment]
Thanks @karencfv . I'm glad we found and can fix that bug.
The delete command is included in these docs examples so I'll merge it after your PR (and as soon as I can figure out how to sign my earlier commits).  :-)

**Comment by kilfoyle** [comment]
Closing and replaced by #284

**Pull request closed without merging**